webapp/src/app/projects/stores/project-invite.state.ts

```diff
@@ -0,0 +1,150 @@
+import { Action, NgxsOnInit, Selector, State, StateContext, Store } from '@ngxs/store';
+import { throwError } from 'rxjs';
+import { catchError, finalize, tap } from 'rxjs/operators';
+import { Logout } from '../../auth/stores/auth.state';
+import { errorToMessage } from '../../shared/util/api-error';
+import { ProjectRole } from '../models/project-role';
+import { ClearCurrentProject, SetCurrentProject } from './projects.state';
+import { ProjectInvite } from '../models/project-invite';
+import { ProjectInviteService } from '../services/project-invite.service';
+
+export class ClearMessages {
+  static readonly type = '[ProjectUser] Clear messages';
+}
+
+export class GetProjectInvites {
+  static readonly type = '[ProjectInvite] Get project invites';
+  constructor(public projectId: string) {}
+}
+
+export class AddProjectInvite {
+  static readonly type = '[ProjectInvite] Add project invite';
+  constructor(public projectId: string, public email: string, public role: ProjectRole) {}
+}
+
+export class UpdateProjectInvite {
+  static readonly type = '[ProjectInvite] Update project invite';
+  constructor(public projectId: string, public inviteId: string, public role: ProjectRole) {}
+}
+
+export class RemoveProjectInvite {
+  static readonly type = '[ProjectInvite] Remove project invite';
+  constructor(public projectId: string, public inviteId: string) {}
+}
+
+export interface ProjectInviteStateModel {
+  invites: ProjectInvite[];
+  isLoading: boolean;
+  errorMessage: string | undefined;
+}
+
+const stateDefaults = {
+  invites: [],
+  isLoading: false,
+  errorMessage: undefined,
+};
+
+@State<ProjectInviteStateModel>({
+  name: 'projectInvites',
+  defaults: stateDefaults,
+})
+export class ProjectInviteState implements NgxsOnInit {
+  constructor(private projectInviteService: ProjectInviteService, private store: Store) {}
+
+  @Selector()
+  static isLoading(state: ProjectInviteStateModel) {
+    return state.isLoading;
+  }
+
+  @Selector()
+  static invites(state: ProjectInviteStateModel) {
+    return state.invites;
+  }
+
+  ngxsOnInit(ctx: StateContext<ProjectInviteStateModel>) {}
+
+  @Action(SetCurrentProject)
+  setCurrentProject(ctx: StateContext<ProjectInviteStateModel>, action: SetCurrentProject) {
```

Comment:
I think you can remove the hook on the setCurrentProject, the invites can be loaded only when the component needs it.

Reply:
Here's were you can dispatch the get invites: https://github.com/traduora/traduora/blob/master/webapp/src/app/projects/components/team-overview/team-overview.component.ts#L37

Reply:
And I think you might still need to add the functions for the update invite and remove invite on the team overview component :)

Reply:
that should be added.